shm/src/Fruit.cpp

```diff
@@ -0,0 +1,20 @@
+#include "shm/inc/Fruit.hpp"
+
+#include <iostream>
+
+Fruit::Fruit(const std::string& name, size_t amount, size_t basePrice) 
+        : Cargo(name, amount, basePrice) {
+}
+
+Fruit& Fruit::operator--() {
+    if (rottenTime !=0){
+        rottenTime--;
+    } else {
+        std::cout << "Fruit is completely rotten\n";
+    }
+    return *this;
+}
+
+std::size_t Fruit::getPrice() {
+    return basePrice_ * (rottenTime / 10);
```

Comment:
Integer division - may lost precision. Tests needed to check the result.

Reply:
Do we need a good precision? If we use this formula everywhere, we shouldn't have any problems